pkg/controller/scyllacluster/resource.go

```diff
@@ -648,17 +650,47 @@ func stringOrDefault(str, def string) string {
 	return def
 }
 
-func MakeServiceAccount(sc *scyllav1.ScyllaCluster) *corev1.ServiceAccount {
+func MakeServiceAccount(sc *scyllav1.ScyllaCluster, serviceAccounts map[string]*corev1.ServiceAccount, serviceAccountLister corev1listers.ServiceAccountLister) (*corev1.ServiceAccount, error) {
+	var annotations map[string]string
+	name := naming.MemberServiceAccountNameForScyllaCluster(sc.Name)
+
+	// In 1.7.0 we started managing ServiceAccounts. Because previously these were created by users
+	// there's a high chance that they don't match our selector, hence are not available in serviceAccounts.
+	//
+	// We need to copy annotations stored in user SA because these may control integrations with k8s provider services
```

Comment:
nit: I'd move this comment over to the block below

Reply:
done

Reply:
I don't see it moved though https://github.com/scylladb/scylla-operator/pull/933/files#diff-d5f374ace56ea95c01835a2bccad226009def5f02e5d88c10e261ef5ee394718R677